projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/KerberosAuthenticationIT.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```

Comment:
```suggestion
/*
 * Copyright 2021 VMware, Inc.
 * SPDX-License-Identifier: Apache-2.0
 */

package com.vmware.taurus.datajobs.it;

import com.vmware.taurus.ServiceApp;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.web.server.LocalServerPort;
import org.springframework.core.Ordered;
import org.springframework.security.kerberos.client.KerberosRestTemplate;
import org.springframework.security.kerberos.test.KerberosSecurityTestcase;
import org.springframework.test.context.TestContext;
import org.springframework.test.context.TestExecutionListener;
import org.springframework.test.context.TestExecutionListeners;
import org.springframework.test.context.TestPropertySource;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import java.io.File;
import java.util.Collections;

@TestPropertySource(properties = {
        "test.dir=target",
        "featureflag.security.enabled=true",
        "datajobs.security.kerberos.enabled=true",
        "datajobs.security.kerberos.kerberosPrincipal=HTTP/localhost",
        "datajobs.security.kerberos.keytabFileLocation=target/foo.keytab",
        "datajobs.security.kerberos.krb5ConfigLocation=target/krb5.conf",
        "java.security.krb5.conf=target/krb5.conf",
        "sun.security.krb5.debug=true"
})
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = ServiceApp.class)
@ExtendWith(SpringExtension.class)
@TestExecutionListeners(value = {KerberosAuthenticationIT.KdcExecutionListener.class}, mergeMode = TestExecutionListeners.MergeMode.MERGE_WITH_DEFAULTS)
public class KerberosAuthenticationIT {
    // composition in favour of inheritance, due to KDC instance is used via TestExecutionListeners and Test both (shared)
    public static final KerberosSecurityTestcase testcase = new KerberosSecurityTestcase();
    public static final File keytab = new File("target/foo.keytab");

    public static class KdcExecutionListener implements TestExecutionListener, Ordered {
        @Override
        public int getOrder() {
            return 0;
        }

        @Override
        public void beforeTestClass(final TestContext testContext) throws Exception {
            testcase.startMiniKdc();
            testcase.getKdc().createPrincipal(keytab, "client/localhost", "HTTP/localhost");
        }
    }

    @LocalServerPort
    int randomPort;

    @Test
    public void test() throws Exception {
        System.out.println();
        KerberosRestTemplate kerberosRestTemplate = new KerberosRestTemplate(keytab.getPath(), "client/localhost", Collections.singletonMap("isInitiator", "true"));
        var response = kerberosRestTemplate.getForObject("http://localhost:" + randomPort + "/data-jobs/for-team/supercollider/jobs", String.class);
    }
}
```
That's a suggestion with bootstrapping KDC before the test using the Spring way. 

Nothing else to configure from KerberosRestTemplate's side - did not find any headers setting. ResponseHeaderSettingKerberosAuthenticationSuccessHandler seems to set some headers to the response upon successful authentication. 
The issue with the test is, that the workflow does not reach KerberosServiceAuthenticationProvider's `authenticate()`.